app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt

```diff
@@ -238,31 +238,47 @@ class VaultRepositoryImpl(
 
         // Setup ciphers MutableStateFlow
         mutableCiphersStateFlow
-            .observeWhenSubscribedAndLoggedIn(authDiskSource.userStateFlow) { activeUserId ->
+            .observeWhenSubscribedAndUnlocked(
+                authDiskSource.userStateFlow,
+                vaultUnlockFlow = vaultUnlockDataStateFlow,
+            ) { activeUserId ->
```

Comment:
The `observeVaultDiskCiphers` already accounts for a locked vault, this logic should not be needed.

Reply:
What is the state the user should be in when the vault is locked but subscribed?

Based on the PR description:
>In some instances the vault gets stuck on an infinite loading.
This is caused by some scenarios when the app is on background and the vault gets locked the subscriptionCount on `observeWhenSubscribedAndLoggedIn` doesn't change causing the observer to not be called anymore.

The flow is placed in a loading state because the user's vault is locked and we are waiting for the vault to be unlocked, this seems to be the desired state.

Reply:
the issue is not the state itself, is that the `observeVaultDiskCiphers` is never called because there is no change on the `observeWhenSubscribedAndLoggedIn` `combine`